### ISSUE TITLE
Map scale info in app

### ISSUE THREAD

**Issue by ollesmaps** [OpenOrienteering/mapper#1040]
Once a map is opened in Anroid app, there is no way how to find out map scale. I think,  it should be written somewhere. You can get a file from stranger, etc.
Maybe a button summarizing scale, grid, declination would do it.

On desktop the only place where to find map scale is at Change map scale? It is worth considering  including it also somewere else. Georeferencing? Map notes?

**Comment by dg0yt** [comment]
Once upon a time the scale was in the georeferencing dialog. I wasn't happy that it was removed.

Now there is a much more "map information" which might be worth displaying, like number of object, symbols etc.

**Comment by lpechacek** [comment]
I tried the possibility of displaying map scale along with zoom level. It was ugly. A separate info dialog looks like a better idea to me.

**Comment by krticka** [comment]
@dg0yt 
> Once upon a time the scale was in the georeferencing dialog. I wasn't happy that it was removed.

As I remember the problem was that the field with scale value was active. Many users made mistake when they accidentally changed it. Or they were changing it intentionally thinking that they will change the scale and also symbol sizes. Than, both mistakes were causing many scale related problems to users.
There is nothing against scale in georeferencing dialog, but field must be inactive.

**Comment by dg0yt** [comment]
> There is nothing against scale in georeferencing dialog, but field must be inactive.

That's what I would have proposed: a read-only field, followed by a button ("Change...", gear, pencil, whatever).

**Comment by krticka** [comment]
There are few other places where it can be placed.
- in georeferencing dialogue (there it was placed previously)
- in change map scale dialogue (info about current scale, very useful here)
- in map notes (there can be generated basic information about map project including scale)

**Comment by Symbian9** [comment]
> I tried the possibility of displaying map scale along with zoom level. It was ugly. A separate info dialog looks like a better idea to me.

Yeah, this info should be somewhere visible, also thought about that.

> Within the app the place could be in info under measurement button.

Fully agreed! Here is draft:

![pic.1](http://i.imgur.com/d5YRvoS.png)

**Comment by dg0yt** [comment]
> > Within the app the place could be in info under measurement button.
>
> Fully agreed!

I disagree. This window is for the current selection (which changes frequently). I don't think it is a good idea to fill it with static information.

**Comment by Symbian9** [comment]
> This window is for the current selection (which changes frequently). I don't think it is a good idea to fill it with static information.

Well, then why not show map scale info here wile there no objects selected?

![pic.2](http://i.imgur.com/SnLQbUq.png)

**Comment by dg0yt** [comment]
That sound feasible, indeed.

**Comment by Symbian9** [comment]
> That sound feasible, indeed.

If so, then maybe show not only map scale info here, but also show other map info (such as georeferencing info, etc.).

So, it would be something like "map dashboard" (**while "No object selected."**)

**Comment by lpechacek** [comment]
FWIW, a new "map info" dialog looks like a the best way to go.

Please note that this issue report started as a request for the mobile UI enhancement. The desktop UI part has been mixed into it along the way.

The mobile UI does not have any georeferencing or map notes dialog. Therefore the map scale info information will have to be presented in a separate dialog or a "toast" and a separate button to trigger the action has to be added. For the desktop UI the same action can be triggered from menu.

I'm not a big fan of adding the information bits into Measure dialog as it looks like a workaround to current mobile UI deficiencies.

A prototype implementation is referenced above.

![image](https://user-images.githubusercontent.com/5584406/47858892-505c1600-dded-11e8-983f-b70673710c63.png)

**Comment by Symbian9** [comment]
> Therefore the map scale info information will have to be presented in a separate dialog or a "toast" and a separate button to trigger the action has to be added. For the desktop UI the same action can be triggered from menu.

Nice implementation, but it's look like add one more widget/notification message window.

For Mobile UI additional notification message window is OK.

But for Desktop UI main idea (that sketched by me above) is to re-use "Measure" widget empty space while there no object selected.

**Comment by krticka** [comment]
Double-tap on Map item in templates seems to be elegant solution but it will be quite hidden solution for many users. What about to add small info icon to Map item?

![clipboard01](https://user-images.githubusercontent.com/15969434/47899821-146f9200-de7b-11e8-9680-1bb25b9e9165.jpg)

**Comment by lpechacek** [comment]
@Symbian9, @krticka, your (intuitive) observation that there is no space left on the main screen is completely correct. However, instead of attempting to make more space on the main screen, you are trying to find a "small change" which has so far led to compromises. I'm pretty sure you feel that few people would would look for that map scale information under _Template setup_ or _Measure object_. 

My proposal is to place  _Map info_ function in the vicinity of _Show whole map_ and _Show grid_ functions in the top left corner. Room for it could be created by hiding less frequently used functions under a single icon. My impression is that at least four of the eight icons in the top left corner are rarely used during a surveying session. They are _Compass display toggle_, _Location display toggle_, _Show whole map_ and _Grid display_. I propose hiding these functions under a new icon, just like the direct skip to 1x and 2x zoom are hidden under _Zoom out_, and adding _Map info_ into that group.

How does that sound to you?

![image](https://user-images.githubusercontent.com/5584406/47902562-944d2a80-de82-11e8-8cfe-3c15db59f000.png)
(Of course, the icon for "Extras" has to be invented. :) )

**Comment by krticka** [comment]
That sounds perfect to me. 
You can put there also Automatic north alignment (personally I am not using this option) and distance rings (I select this option only once when I open the file and start mapping). Then there will be more space for the other editing functions on smaller screens.

**Comment by krticka** [comment]
Relates to #865

**Comment by ollesmaps** [comment]
@lpechacek I agree, it should have a separate button.

However I would like to keep _Show whole map_ visible as I use it quite often (need to get an overview what had been done and needs to be done).  I would not mind though  if _Show whole map_ gets moved in the down left corner, where zooming occurs. It very much relates - after maxi zoom out you have to zoom in. As suggested in #952.

Could the _Map Info_  also contain map statistics as suggested in #1097?

**Comment by Symbian9** [comment]
> What about to add small info icon to Map item?

Seems like good solution too. Just move "Info" button to the bottom of "Templates" widget:

![pic.1](https://i.imgur.com/kPkKzBj.png)

**Comment by Symbian9** [comment]
> Could the Map Info also contain map statistics

+1. This is what I mean under *"map dashboard"* inside "Measure" widget (in [my comment](https://github.com/OpenOrienteering/mapper/issues/1040#issuecomment-434851593) above)

**Comment by Symbian9** [comment]
> seems to me better as it fixes problems with accessibility of some editing functions on smaller screens.

This is Mobile UI issue, and for improve Mobile UI we need separate issue thread. Current Mobile UI is not yet complete and should be redesigned.
